### DISCUSSION TITLE
Transfer and synchronize playlists between providers

### DISCUSSION THREAD

**Discussion by lwsx** [music-assistant/support#2376] [Feature requests and ideas]
**What?** 
Ability for the Music Assistant server to move and sync playlists between different providers

**Why?**
While playlist transfer and sync providers exist, there are no current ones that may be self-hosted, and they require recurring subscriptions

---

Since there is the ability to link multiple providers to the Music Assistant server and it has the potential to edit content within them, as well as knowing which song is which through the metadata provider connection, I feel like it would be possible to have the server synchronize playlists between services or transfer them from one to another. 

There are multiple services such as this available commercially on the internet (Soundiiz, tunemymusic), however they are all paid and require a recurring subscription. Playlist synchronization would allow for more abstraction of the provider services, and position Music Assistant as a more universal music library management solution, instead of just a part of it.

**Comment by OzGav** [comment]
You can transfer a play list between providers by opening the playlist, selecting all the tracks and then ADD TO PLAYLIST for the new provider.

**Reply by malcolmrigg** [reply]
That's a reasonable work around for small playlists - but I can't figure out how to do a "select all". Is this possible currently?

I've got a couple of large (500+ song) playlists and don't have the patience to go through and select them all!

**Reply by OzGav** [reply]
Use CTRL-A

**Reply by malcolmrigg** [reply]
*facepalm* of course, thank you!

**Comment by snds** [comment]
I don't see any way to do this with Apple Music. I've connected it using the music token, but all I get synced are artists & albums. I'd like to be able to use my playlists in the assistant if possible

**Reply by OzGav** [reply]
Playlists are supported but note this from the docs 

> MA can only play an uploaded track if Apple has linked it to an online version in the Apple Music catalog. If that link is not found, MA will not do the import.

**Comment by stefjuh1995** [comment]
Would still be interested in a way to automatically keep playlists synchronized between different providers.

**Reply by OzGav** [reply]
It is not automatic but the new export and import function should make the process fairly easy. https://beta.music-assistant.io/usage/#importing-and-exporting
